solutions/cses-1733.mdx

```diff
@@ -62,4 +62,64 @@ int main() {
 ```
 
 </CPPSection>
+	
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+ 
+public class Main {
+	public static long p = 69420;
+	public static long M = (long)(1e9 + 7);
+	
+	public static long[] basePow;
+	public static long[] pref;
+ 
+	public static long getHash(int l, int r) {
+		long h = pref[r + 1] - (basePow[r - l + 1] * pref[l] % M) % M;
+		return h < 0 ? h + M : h;
+	}
+ 
+public static void main(String[] args) {
+	Kattio io = new Kattio();
+
+	char[] s = io.next().toCharArray();
+	int n = s.length;
+		
+	basePow = new long[n + 1];
+	basePow[0] = 1;
+	for(int i = 1; i <= n; i++) {
+		basePow[i] = basePow[i - 1] * p % M;
+	}
+ 
+	pref = new long[n + 1];
+	for(int i = 1; i <= n; i++) {
+		pref[i] = (pref[i - 1] * p + s[i - 1]) % M;
+	}
+ 
+	for(int i = 0;i<n;i++) {
+		int curridx = 0;
+		boolean ok = true;
+		while(curridx < n) {
+		int len = Math.min(i + 1, n - curridx);
+		ok &= (getHash(0, len-1) == getHash(curridx, curridx + len - 1));
+			curridx += len;
+		}
+ 
+		if(ok) {
+			io.print((i + 1) + " ");
+		}
+	}
+		
+	io.close();
+	}
```

Comment:
what's w/ the indentation here

Reply:
Hi, could you tell me the specific places where it seems off?
And, I just noticed that I have made a grave mistake I realised that I have a different template which is very similar to this code and I never tested this code out on CSES until now. And, This code now TLE's and the code with my other template tends to get accepted on it. Do you recommend I completely close this PR and fully update my solution with an better alternative for Java IO to make it pass on CSES?

P.S. The Approach is the exact same but the TLE is caused by the Input Output part mainly

Reply:
sure

Reply:
Do you recommend I close down the PR and make a fresh one? And I am very sorry for any inconveniences caused

Reply:
sure